pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -114,8 +114,7 @@ func (m *Monitor) handleOSDMarkedOut(outOSDid int) error {
 		logger.Infof("osd.%d is 'safe-to-destroy'", outOSDid)
 		label := fmt.Sprintf("ceph-osd-id=%d", outOSDid)
 		dp, _ := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
-
-		if dp.Items != nil {
+		if len(dp.Items) != 0 {
```

Comment:
I think this may be the wrong place/method to avoid a panic. To me it seems that `dp.Items != nil` will only panic if `dp` itself is `nil`. Really, the error from this should not have been masked in the first place, and there should be an `if err != nil` afterwards. After checking the error, it should be safe to check `len(dp.Items)`

Reply:
As we are just deleting the out OSDs deployment, the out OSD will still be there, so every time the deployment of out OSD is fetched, it will give error. Is this desirable?

Reply:
@rohan47 To echo what @BlaineEXE is saying, let's check for an err from `GetDeployments()` to avoid the crashing condition. We have this pattern of checking the err for IsNotFound in many places. In this case, I would recommend:
```
dp, err := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
if err != nil {
  if errors.IsNotFound(err) {
    return nil // nothing to remove
  }
  return fmt.Errorf("failed to get deployments. %+v", err)
}
if len(dp.Items) > 0 {
  // delete the deployment
}
```

Reply:
Got it, thanks.
Updated the PR